app/modules/code_provider/provider_factory.py

```
@@ -0,0 +1,454 @@
+
+        # Create provider instance
+        if provider_type == ProviderType.GITHUB:
+            base_url = base_url or "https://api.github.com"
+            provider = GitHubProvider(base_url=base_url)
+
+        elif provider_type == ProviderType.GITBUCKET:
+            if not base_url:
+                raise ValueError(
+                    "GitBucket requires CODE_PROVIDER_BASE_URL environment variable. "
+                    "Example: CODE_PROVIDER_BASE_URL=http://localhost:8080/api/v3"
+                )
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)